qiskit/providers/ibmq/api/clients/auth.py

```diff
@@ -75,12 +75,12 @@ def _request_access_token(self) -> str:
             response = self.client_auth.login(self.api_token)
             return response['id']
         except RequestsApiError as ex:
-            response_obj = ex.original_exception.response
-            if response_obj is not None and response_obj.status_code == 401:
+            error_response = ex.__cause__.response    # pylint: disable=no-member
+            if error_response is not None and error_response.status_code == 401:
```

Comment:
Can you revise a bit the variable and the `if`?

Previously, we had strong assurances that `ex.original_exception` was coming from `requests` - now we still have them, although we are open to the more general case of the exception being `raise from` somewhere else (although in practice it doesn't seem it will happen yet). Can we do some defensive programming and try something along the lines of:
```
if isinstance(ex.__cause__, Blah) and ...
```